include/appbase/application.hpp

```diff
@@ -341,4 +360,17 @@ namespace appbase {
       }
    }
 
+   class scoped_app {
+   public:
+      scoped_app() : app_(app()) {}
+      ~scoped_app() { application::reset_app_singleton(); } // destroy app instance so next instance gets a clean one
+
+      // access methods
+      application*       operator->()       { return &app_; }
+      const application* operator->() const { return &app_; }
+
+   private:
+      application& app_;
+   };
```

Comment:
It's unfortunate to see more contortions to avoid making `application` non-static

Reply:
Yes, I agree that it would be cleaner if the application was not a static. 

On the other hand, it would be less convenient as plugins could not just call `app()` to get the app instance, but it would have to be captured or passed through. 

I think this change is already pretty big, and removing the application singleton would be a much bigger change, so if we want to do this, it might be better to do it in a separate PR.

Reply:
This is fine, but you could have just used `std::unique_ptr`.

Reply:
Yes. I think this is a little cleaner, as it `assert(application::null_app_singleton());` in the constructor and calls `application::reset_app_singleton();` in the destructor. With unique_ptr I would have had to define a `Deleter` type anyways.

Reply:
Thanks Kevin for the review and approval, much appreciated! I just added a test to make sure that queued items are discarded when `app->quit()` is received.